libbroker/broker/internal/peering.cc

```diff
@@ -119,9 +122,10 @@ void peering::on_bye_ack() {
   bye_timeout_.dispose();
 }
 
-void peering::force_disconnect() {
+void peering::force_disconnect(const std::string& reason) {
   if (!removed_) {
     removed_ = true;
+    removed_reason_ = reason;
```

Comment:
Always forces copies, better:

```suggestion
void peering::force_disconnect(std::string reason) {
  if (!removed_) {
    removed_ = true;
    removed_reason_ = std::move(reason);
```